[12.0][MIG] web_translate_dialog + improvements

tschanzt opened the pull request.

Commit:
[CHG] rename web_translate_dialog_page to web_translate_dialog

Commit:
[FIX] autosize the textareas

Commit:
[CHG] read_translations() not working as expected on OpenERP 7, use the normal read()

Commit:
[FIX] error when a view has no sidebar (popup)

Commit:
[FIX] avoid to set the dirty flag on the view when not necessary, especially when the translate dialog is opened from the readonly view.

Commit:
[FIX] display title

Commit:
[ADD] pot file

Commit:
[IMP] Translations files

Commit:
[ADD] fr translation

Commit:
Rename the frenchisms 'word' trads to translation

Commit:
Launchpad automatic translations update.

Commit:
Launchpad automatic translations update.

Commit:
Launchpad automatic translations update.

Commit:
obsolete comment

Commit:
Read the fields of the current language from the view, not from a read() in the database

Commit:
Propagate the changes done in the translate dialog back to the view

Commit:
Launchpad automatic translations update.

Commit:
Launchpad automatic translations update.

Commit:
Launchpad automatic translations update.

Commit:
Launchpad automatic translations update.

hbrunn committed:
move all addons to __unported__, set all addons as not installable

alexis-via committed:
web_translate_dialog: port to v8

alexis-via committed:
Move module description to README.rst

alexis-via committed:
Improve README

alexis-via committed:
web_translate_dialog: Add author and license key in __openerp__.py

alexis-via committed:
web_translate_dialog: update module category

yvaucher committed:
Add missing default oca icons

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

oca-transbot committed:
OCA Transbot updated translations from Transifex

gurneyalex committed:
minor improvements to README.rst

oca-transbot committed:
OCA Transbot updated translations from Transifex

antespi committed:
[MIG] web_translate_dialog

antespi committed:
[FIX] Add codeview button to translate HTML fields

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

oca-transbot committed:
OCA Transbot updated translations from Transifex

JayVora-SerpentCS committed:
[MIG] web_translate_dialog migrated to v10

Commit:
[fix] web_translate_dialog: correct context split

Fixes the display of languages other than the user language  in the translation dialog.
split_context checks if we provide more arguments than read has and then uses of the additional one as context.
Since read takes the argument load additionally to the ones already given, we need to provide it as well to ensure the context is split.

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update web_translate_dialog.pot

Commit:
[12.0][MIG] web_translate_dialog

pedrobaeza commented:
Latest versions of this module were broken due to multiple ir.translation records when html translate was used. Have you taken this into account? And I think there's a rework in master about this. Maybe you can check it to see if it's worth to backport it.

tschanzt commented:
@pedrobaeza No not actively, as i wasn't aware of the issue. But do you have more details on the issue maybe i can check it out.

pedrobaeza commented:
#835

tschanzt commented:
Well that part is rewritten a bit. But the technical details in #844 which tries to fix the same issue mentioned in #835 is because context isn't passed in kwargs which it now is.

pedrobaeza commented:
OK, please check Travis status and I'll give it a try to see.

Commit:
Fix issue in manifest & move to readme folder

yvaucher commented on web_translate_dialog/static/src/js/web_translate_dialog.js:
```diff
@@ -0,0 +1,224 @@
+
+        this.$el.find('.oe_translation_field').val('').removeClass('touched');
+        _.each(self.languages, function(lg) {
+            var context = session.user_context;
```
As discussed together, we want to change this line to create a new context for each language.

yvaucher reviewed (commented):
Code review

LGTM, waiting for the change on session before approving.

simahawk commented on web_translate_dialog/__manifest__.py:
```diff
@@ -0,0 +1,26 @@
+    "summary": "Easy-to-use pop-up to translate fields in several languages",
+    "version": "12.0.1.0.0",
+    "category": "Web",
+    "website": "https://odoo-community.org/",
```
https://github.com/OCA/web

Commit:
fix context and add mutex

Tardo reviewed (changes requested):
If uses suggestion link gets this error:
![error_translation](https://user-images.githubusercontent.com/731270/60221254-5372ad80-9879-11e9-90b1-5aea3d8a8c38.gif)

Commit:
fix reminder box link

tschanzt commented:
@Tardo i updated the pr. Should work now. 
@pedrobaeza do you want to give it a try as well

pedrobaeza reviewed (changes requested):
Hi, I have been trying the module in runbot, and there's a weird behavior. I have loaded 2 languages: the English standard one + Spanish. Having the interface in Spanish, look what happens:

![Peek 28-06-2019 16-55](https://user-images.githubusercontent.com/7165771/60351288-bd986900-99c5-11e9-8716-bc85c9b95de0.gif)

grindtildeath committed:
Apply autoresize on textarea elements and fix css

grindtildeath commented:
Small fixes: https://github.com/tschanzt/web/pull/1

grindtildeath committed:
Fix error if field to translate is not textarea

grindtildeath committed:
Keep translated value after hitting save

grindtildeath commented:
@tschanzt Small fixes here : https://github.com/tschanzt/web/pull/2

grindtildeath commented:
@pedrobaeza your issue is fixed in 2de0212  :wink:

grindtildeath commented:
@Tardo can you please update your review ?

Tardo commented on web_translate_dialog/static/src/js/web_translate_dialog.js:
```diff
@@ -0,0 +1,232 @@
+
+var translateDialog = Dialog.extend({
+    template: "TranslateDialog",
+    init: function(parent, event_data, res_id, content) {
```
'content' is really used?

Tardo commented on web_translate_dialog/static/src/js/web_translate_dialog.js:
```diff
@@ -0,0 +1,232 @@
+    },
+});
+
+View.include({
```
:fire:

Tardo commented on web_translate_dialog/static/src/js/web_translate_dialog.js:
```diff
@@ -0,0 +1,232 @@
+});
+
+View.include({
+});
```
:fire:

Tardo commented on web_translate_dialog/static/src/xml/base.xml:
```diff
@@ -0,0 +1,53 @@
+                      <label class="oe_label"><t t-esc="field.string"/>:</label>
+                  </td>
+                  <td t-foreach="widget.languages" t-as="lg" class="oe_form_frame_cell">
+                      <input t-if="field.type == 'char' || field.type == 'url'"
```
```suggestion
                      <input t-if="['char','url'].indexOf(field.type) !== -1"
```

Tardo commented on web_translate_dialog/static/src/xml/base.xml:
```diff
@@ -0,0 +1,53 @@
+                      <input t-if="field.type == 'char' || field.type == 'url'"
+                             type="text" t-attf-name="#{lg.code}-#{field_name}"
+                             value="" data-value="" class="oe_translation_field"/>
+                      <textarea t-if="field.type == 'text'"
```
```suggestion
                      <textarea t-elif="field.type === 'text'"
```

Tardo commented on web_translate_dialog/static/src/xml/base.xml:
```diff
@@ -0,0 +1,53 @@
+                      <textarea t-if="field.type == 'text'"
+                                t-attf-name="#{lg.code}-#{field_name}" data-value=""
+                                class="oe_translation_field" ></textarea>
+                      <div t-if="field.type == 'html'" class="oe_form_field_html">
```
```suggestion
                      <div t-elif="field.type === 'html'" class="oe_form_field_html">
```

grindtildeath committed:
Refactor to display only existing translations

grindtildeath committed:
Attend review comments

TDu commented:
:+1:

Tardo commented on web_translate_dialog/static/src/js/web_translate_dialog.js:
```diff
@@ -0,0 +1,224 @@
+    init: function(parent, options) {
+        this._super(parent,
+                    {title: _t("Translations"),
+                     width: '90%',
```
I can't see where 'width' and 'height' are used...

Tardo commented on web_translate_dialog/static/src/js/web_translate_dialog.js:
```diff
@@ -0,0 +1,224 @@
+    },
+    start: function() {
+        var self = this;
+        this.$el.find('.oe_translation_field').change(function() {
```
All widgets have "$" method that call to 'this.$el.find'. You can replace all ```this.$el.find(``` by ```this.$(```

Tonow-c2c commented on web_translate_dialog/static/src/xml/base.xml:
```diff
@@ -0,0 +1,52 @@
+        <div class="modal-body">
+            <table t-if="widget.translatable_field"
+                   class="oe_frame oe_forms oe_translation_form"
+                   border="0" cellpadding="0" cellspacing="0" width="100%">
```
in the future it's will be great to create a less file and remove style on xml

grindtildeath committed:
Set max-width to 90% on translate dialog

grindtildeath committed:
Use $ instead of $el.find

grindtildeath committed:
Fix sizing of textarea

Resize textarea either to the max height of its content if it stays
in the modal or to the max height available in the modal

grindtildeath committed:
Remove field description column and move it to title

simahawk commented on web_translate_dialog/static/src/js/web_translate_dialog.js:
```diff
@@ -71,6 +71,22 @@ var translateDialog = Dialog.extend({
+        var max_height = 100
+        // Resize textarea either to the max height of its content if it stays
+        // in the modal or to the max height available in the modal
+        if (textareas !== undefined && textareas[0] !== undefined) {
```
just check `$textareas.length`

simahawk commented:
just a small chance on OCA PR, but looks good

grindtildeath committed:
Add tr_sequence on res.lang to allow sorting languages in the translation dialog

nikul-serpentcs commented on web_translate_dialog/models/res_lang.py:
```diff
@@ -0,0 +1,19 @@
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        print(order)
```
@tschanzt Removed `print`

grindtildeath replied:
Whole extension of search is unneeded, thanks for the fast feedback :wink:

grindtildeath commented:
@pedrobaeza @Tardo @simahawk @Tonow-c2c @nikul-serpentcs 
Can you guys please update your reviews before we rebase and squash ?

grindtildeath commented:
@pedrobaeza @Tardo @simahawk @Tonow-c2c @nikul-serpentcs
ping, let's get this merged please :pray:

pedrobaeza commented:
Sorry, not able to review now

simahawk commented on web_translate_dialog/static/src/js/web_translate_dialog.js:
```diff
@@ -0,0 +1,241 @@
+var QWeb = core.qweb;
+var Mutex = concurrency.Mutex;
+
+var translateDialog = Dialog.extend({
```
TranslateDialog

simahawk commented on web_translate_dialog/static/src/js/web_translate_dialog.js:
```diff
@@ -0,0 +1,241 @@
+
+        this.$('.oe_translation_field').val('').removeClass('touched');
+
+        var deff = $.Deferred();
```
def

simahawk commented on web_translate_dialog/static/src/xml/base.xml:
```diff
@@ -0,0 +1,46 @@
+                </th>
+            </tr>
+            <t t-value="widget.translatable_field" t-set="field_name" />
+              <t t-set="field" t-value="widget.view.searchView.fields[field_name]" />
```
bad indent

simahawk commented on web_translate_dialog/models/base.py:
```diff
@@ -0,0 +1,34 @@
+                ('res_id', '=', rec_id)
+            ])
+            for tr_res in tr_read_res:
+                if tr_res.get('lang') == 'en_US':
```
this could be excluded in the query

simahawk reviewed (approved):
LG, just few non-blocking comments.

simahawk commented:
@grindtildeath please squash some commits :)

Commit:
address comments

tschanzt commented:
@simahawk updated and i squashed some commits

daramousk commented on web_translate_dialog/models/base.py:
```diff
@@ -0,0 +1,33 @@
+
+class BaseModel(models.BaseModel):
+
+    _inherit = 'base'
```
No need for this:

https://github.com/OCA/OCB/blob/12.0/odoo/models.py#L434

simahawk replied:
that's added only when current name is not `base`. In any case: explicit it's better than implicit :wink:

daramousk commented on web_translate_dialog/static/src/js/web_translate_dialog.js:
```diff
@@ -0,0 +1,241 @@
+});
+
+FormView.include({
+    render_sidebar: function($node) {
```
This does not seem to exist on the `FormView`, therefore not called anywhere.

simahawk replied:
@tschanzt it's `renderSidebar` indeed. Do you have any piece of code defining `render_sidebar`?

daramousk commented on web_translate_dialog/readme/USAGE.rst:
```diff
@@ -0,0 +1,8 @@
+Go to an object that has translatable fields (*Products* for example) and
+select *More > Translate* (or click on *Edit* and then click on the flag at
```
I do not see a `More` button here as well.

simahawk reviewed (changes requested):
form sidebar needs a fix

Commit:
[IMP] web_translate_dialog: set maxlength in dialog

when field support size attribute set it to dialog to prevent aditional
input

Commit:
Fix sidebar & reenable multiple fields

tschanzt commented:
@simahawk @daramousk i updated the sidebar stuff and readded possibility to actually use that function

simahawk commented:
@daramousk would you mind to pass by again?

MeetKD commented on web_translate_dialog/static/src/xml/base.xml:
```diff
@@ -0,0 +1,53 @@
+  </t>
+
+  <t t-name="TranslateDialog.buttons">
+    <button class="btn btn-sm oe_button btn-primary oe_form_translate_dialog_save_button">
```
Can we remove oe_button here and btn-primary class.

MeetKD commented on web_translate_dialog/static/src/xml/base.xml:
```diff
@@ -0,0 +1,53 @@
+      <span>Save</span>
+    </button>
+    <button class="btn btn-sm oe_button btn-default oe_form_translate_dialog_cancel_button">
+      <span>Cancel</span>
```
Same her.e btn-secondary Class may be ?

MeetKD commented on web_translate_dialog/static/src/xml/base.xml:
```diff
@@ -0,0 +1,53 @@
+      </t>
+      </table>
+    </div>
+  </t>
```
I would also suggest to replace all the oe instance like oe_label, oe_form_field etc.

MeetKD reviewed (commented):
Some non-blocking improvements

yvaucher commented:
/ocabot merge

yvaucher commented:
@simahawk @pedrobaeza I reseted your review, but seems we have enough approval.

By the way @tschanzt doesn't work anymore at C2C, so I propose to proceed with this PR and make improvement in further PRs.
